### PR TITLE
Added commands and renamed !img

### LINKED ISSUE #10: Move non mod commands out of PortAuthority to Lighthouse

### PR DESCRIPTION
**Pull request by AJMC-DEV**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "canvas": "^2.7.0",
     "discord.js": "^12.5.1",
     "dotenv": "^8.2.0",
+    "package.json": "^2.0.1",
```
Comment (ckohen) [inline package.json]: What's this for?

File: src/rafts/lighthouse/commands/abstract.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
 'use strict';
 /* We did this without ck's help */
 /* eslint-disable-next-line no-unused-vars */
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/abstract.js]: there's no unused vars on the next line
```suggestion
```

File: src/rafts/lighthouse/commands/abstract.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/abstract.js]: Avoid disabling eslint rules, especially for entire files, and prettier should never be disabled
```suggestion
```

File: src/rafts/lighthouse/commands/fractal.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable prettier/prettier */
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/fractal.js]:
```suggestion
```

File: src/rafts/lighthouse/commands/fractal.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+'use strict';
+/* We did this without ck's help */
+/* eslint-disable-next-line no-unused-vars */
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/fractal.js]:
```suggestion
```

File: src/rafts/lighthouse/commands/fractal.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+const Discord = require('discord.js');
+
+const BaseCommand = require('../../BaseCommand');
+
+class FractalCommand extends BaseCommand {
+    constructor(boat) {
+        const options = {
+            name: 'fractal',
+            owner: true,
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/fractal.js]: Should this command be owner only?

File: src/rafts/lighthouse/commands/abstract.js [hunk 7]
```diff
@@ -27,10 +28,18 @@ class ImgCommand extends BaseCommand {
 
     context.fillStyle = '#57C7FF';
     context.fillRect(0, 0, width, height);
-    const attachment = new Discord.MessageAttachment(canvas.toBuffer(), 'welcome-image.png');
+    var i;
+    var maxIterations = 20;
+    for (i = 0; i < maxIterations; i++) {
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/abstract.js]:
```suggestion
    const maxIterations = 20;
    for (const i = 0; i < maxIterations; i++) {
```
Condense this down, also, might consider taking maxIterations as an argument for this command?

No vars!!!!

File: src/rafts/lighthouse/commands/fractal.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            enabled: true,
+        };
+        super(boat, options);
+    }
+
+    run(message) {
+        const width = 1200;
+        const height = 1200;
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/fractal.js]: Maybe take these as options for the command

File: src/rafts/lighthouse/commands/fractal.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+        function checkIfBelongsToMandelbrotSet(x, y) {
+            var realComponentOfResult = x;
+            var imaginaryComponentOfResult = y;
+            var maxIterations = 300;
+            for (var i = 0; i < maxIterations; i++) {
+                var tempRealComponent = realComponentOfResult * realComponentOfResult
+                    - imaginaryComponentOfResult * imaginaryComponentOfResult
+                    + x;
+                var tempImaginaryComponent = 2 * realComponentOfResult * imaginaryComponentOfResult
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/fractal.js]: No vars, change these to be let or const as necessary, prefer const if the values don't change

File: src/rafts/lighthouse/commands/fractal.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            // Return zero if in set
+            return 0;
+        }
+
+        var magnificationFactor = 2000;
+        var panX = (Math.random() * 2);
+        var panY = (Math.random() * 1);
+        for (var x = 0; x < canvas.width; x++) {
+            for (var y = 0; y < canvas.height; y++) {
+                var belongsToSet =
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/fractal.js]: No vars

File: src/rafts/lighthouse/commands/stars.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
+'use strict';
+/* We did this without ck's help */
+/* eslint-disable-next-line no-unused-vars */
+const util = require('util');
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/stars.js]: not using this, remove it

File: src/rafts/lighthouse/commands/stars.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+      name: 'stars',
+      owner: false,
+      enabled: true,
+    };
+    super(boat, options);
+  }
+
+  run(message) {
+    const width = 1200;
+    const height = 730;
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/stars.js:21]: Input params again?

File: src/rafts/lighthouse/commands/stars.js [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    const canvas = createCanvas(width, height);
+    const context = canvas.getContext('2d');
+
+    const stars = Math.floor(Math.random() * 101) + 75;
+    const lines = Math.floor(Math.random() * 5) + 2;
+    var a = [];
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/stars.js]: ...

File: src/rafts/lighthouse/commands/stars.js [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    context.fillStyle = 'white';
+    context.strokeStyle = 'white';
+    context.lineJoin = 'miter';
+    context.lineWidth = 5;
+
+    for (var i = 0; i <= stars; i++) {
+      var x = Math.random(i) * 1200;
+      var y = Math.random(i) * 730;
+      a.push({ x: x, y: y });
+      var size = (Math.floor(Math.random() * 20) + 10) / 10;
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/stars.js]: You know what I'm going to say

File: src/rafts/lighthouse/commands/stars.js [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+    context.fillStyle = 'black';
+    context.fillRect(0, 0, width, height);
+    context.fillStyle = 'white';
+    context.strokeStyle = 'white';
+    context.lineJoin = 'miter';
+    context.lineWidth = 5;
+
+    for (var i = 0; i <= stars; i++) {
+      var x = Math.random(i) * 1200;
+      var y = Math.random(i) * 730;
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/stars.js]: Use your height and width variables

File: src/rafts/lighthouse/commands/tree.js [hunk 16]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable prettier/prettier */
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/tree.js]: same as above

File: src/rafts/lighthouse/commands/tree.js [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+'use strict';
+/* We did this without ck's help */
+/* eslint-disable-next-line no-unused-vars */
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/tree.js]: same as above

File: src/rafts/lighthouse/commands/tree.js [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+
+const Discord = require('discord.js');
+
+const BaseCommand = require('../../BaseCommand');
+
+class TreeCommand extends BaseCommand {
+    constructor(boat) {
+        const options = {
+            name: 'tree',
+            owner: true,
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/tree.js]: same

File: src/rafts/lighthouse/commands/tree.js [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+        super(boat, options);
+    }
+
+    run(message) {
+        const width = 600;
+        const height = 600;
+        var canvas = createCanvas(width, height);
+        const ctx = canvas.getContext('2d');
+        let curve = 10;
+        let curve2 = 0;
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/tree.js]: params? no var

File: src/rafts/lighthouse/commands/tree.js [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        // StartX, startY, len, angle, branchWidth, color1, color2
+        let centerPointX = canvas.width / 2;
+        let len = Math.floor((Math.random() * 20) + 140);
+        let angle = 0;
+        let branchWidth = (Math.random() * 70) + 1;
+        let color1 = `rgb(${Math.floor(Math.random() * 256)},${Math.floor(Math.random() * 256)},${Math.floor(Math.random() * 256)})`;
+        let color2 = `rgb(${Math.floor(Math.random() * 256)},${Math.floor(Math.random() * 256)},${Math.floor(Math.random() * 256)})`;
```
Comment (ckohen) [inline src/rafts/lighthouse/commands/tree.js]: these can all be const if I'm not mistaken

File: src/rafts/lighthouse/index.js [hunk 21]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const { Collection } = require('discord.js');
+const commands = require('./commands');
+const util = require('../../util');
+const BaseRaft = require('../BaseRaft');
+
+/**
+ * The management raft for this boat.
```
Comment (ckohen) [inline src/rafts/lighthouse/index.js]: Change this description

**Review by ckohen — CHANGES_REQUESTED** [review]

File: src/rafts/lighthouse/commands/fractal.js (hunk 8 above)
Comment (Markens123) [reply src/rafts/lighthouse/commands/fractal.js]: atm no but maybe sometime in the future

**Review by ckohen — APPROVED** [review]

**Pull request merged**